scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -682,35 +682,31 @@ def _response_constraints_validation(self, response, result):
         check_type = PostProcessCheckType.CONSTRAINT_VALIDATION
         error_success = 'Constraints check passed'
         error_failure = 'Constraints check failed'
-        error_name_does_not_exist = 'Constraint check expected a value named "{name}" but it does not exists in the response."'
+        error_name_does_not_exist = 'Constraint check expected a value named "{name}" but it does not exist in the response."'
 
         for value in self.response['values']:
             if 'constraints' not in value:
                 continue
 
-            ensure_has_value_constraint_check = False
-            expected_name = 'value'
             received_value = response.get('value')
             if not self.is_attribute:
                 expected_name = value.get('name')
                 if received_value is None or expected_name not in received_value:
-                    # It is possible we are checking for constraint where
-                    # `hasValue: false`.
-                    ensure_has_value_constraint_check = True
+                    # Since we did not get any value we are explicitly the only case where we want
+                    # to proceed with the constraint check and not error out is if there is a
+                    # constraint that is explicitly checking to see if there is not value provided.
+                    if 'hasValue' not in value['constraints']:
+                        result.error(check_type, error_name_does_not_exist.format(
+                            name=expected_name))
+                        continue
```

Comment:
I'm still not following this.  Why do we need more logic here than just setting `received_value` to `None`?  What is an example case where this check would be expected to fail vs succeed?

Reply:
The best hypothetical example is perhaps something like constraint `notValue: 254`, while it is true that it is not that value, it is likely silently giving the test writer a false sense that a value is coming in just that it is not 254. While this is technically correct, I would argue that `hasValue: false` here to be more explicit of a check that should be happening if someone is apply a constraint to the result